src/utils.jl

```diff
@@ -54,3 +54,55 @@ function checkbalanced(x::AbstractVecOrMat, y::AbstractVecOrMat)
         throw(ArgumentError("source and target marginals are not balanced"))
     return nothing
 end
+
+struct FiniteDiscreteMeasure{X<:AbstractVector,P<:AbstractVector}
+    support::X
+    p::P
+
+    function FiniteDiscreteMeasure{X,P}(support::X, p::P) where {X,P}
+        length(support) == length(p) || error("length of `support` and `p` must be equal")
+        isprobvec(p) ≈ 1 || error("`p` must be a probability vector")
```

Comment:
I guess this should be

```suggestion
        isprobvec(p) || error("`p` must be a probability vector")
```
?